### DISCUSSION TITLE
Problem with restoring the saved state and retrain from there in a new script

### DISCUSSION THREAD

**Discussion by LeeStr1** [lululxvi/deepxde#1891] [Q&A]
This is how I am saving the trained states:

```
net = dde.nn.FNN([2] + [50] * 7 + [4], "swish", "Glorot normal")
model = dde.Model(data, net)
model.compile("adam", lr=0.001)
checkpointer = dde.callbacks.ModelCheckpoint(
    "model/prandtle10", verbose=1, save_better_only=True
)

# Train the model
loss_history, train_state = model.train(epochs=20000, callbacks=[checkpointer])
```

This is how I am trying to restore the previously trained state in a new code and trying to retrain from there (with changed input parameters):

```
net = dde.nn.FNN([2] + [50] * 7 + [4], "swish", "Glorot normal")
model = dde.Model(data, net)

model.restore("model/prandtle10-3000.ckpt", verbose=1)

model.compile("adam", lr=0.001)

checkpointer = dde.callbacks.ModelCheckpoint(
    "model/prandtle5", verbose=1, save_better_only=True
)

# Train the model
loss_history, train_state = model.train(epochs=20000, callbacks=[checkpointer])
```

But, I get following errors:

> AttributeError                            Traceback (most recent call last)
> Cell In[10], line 93
>      90 net = dde.nn.FNN([2] + [50] * 7 + [4], "swish", "Glorot normal")
>      91 model = dde.Model(data, net)
> ---> 93 model.restore("model/prandtle10-3000.ckpt", verbose=1)
>      95 model.compile("adam", lr=0.001)
>      97 checkpointer = dde.callbacks.ModelCheckpoint(
>      98     "model/prandtle5", verbose=1, save_better_only=True
>      99 )
> 
> File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\deepxde\model.py:1089, in Model.restore(self, save_path, device, verbose)
>    1087     print("Restoring model from {} ...\n".format(save_path))
>    1088 if backend_name == "tensorflow.compat.v1":
> -> 1089     self.saver.restore(self.sess, save_path)
>    1090 elif backend_name == "tensorflow":
>    1091     self.net.load_weights(save_path)
> 
> AttributeError: 'NoneType' object has no attribute 'restore'

**Comment by lululxvi** [comment]
```
model.compile("adam", lr=0.001)
model.restore("model/prandtle10-3000.ckpt", verbose=1)
```
